crates/y-sweet/src/server.rs

```diff
@@ -265,8 +266,16 @@ async fn auth_doc(
         None
     };
 
-    let schema = if server_state.use_https { "wss" } else { "ws" };
-    let url = format!("{schema}://{host}/doc/ws");
+    let url = if let Some(url_prefix) = &server_state.url_prefix {
+        let mut url = url_prefix.clone();
+        let scheme = if url.scheme() == "https" { "wss" } else { "ws" };
+        url.set_scheme(scheme).unwrap();
+        url.join("/doc/ws").unwrap().to_string();
+        url.to_string()
+    } else {
+        format!("ws://{host}/doc/ws")
```

Comment:
This `url` is used by the client, right? If so, will running the server in prod essentially mean always passing a `url_prefix`? Assuming that running in prod will always mean using HTTPS.

Reply:
Yes. Maybe if `--prod` is passed without a `--url-prefix`, we should print a warning or even fail to start the server?

Reply:
Maybe it makes sense to disentangle the two options and have the url prefix just be a path prefix? And then HTTPS is on by default unless you pass `--http` or something like that?

Reply:
Yeah, true